mne/stats/regression.py

```diff
@@ -134,3 +136,193 @@ def _fit_lm(data, design_matrix, names):
+        done. If dict, keys are types ('grad' | 'mag' | 'eeg' | 'eog' | 'ecg')
+        and values are the maximal peak-to-peak values to select rejected
+        epochs. E.g. reject = dict(grad=4000e-12, # T / m (gradiometers)
+                                   mag=4e-11, # T (magnetometers)
+                                   eeg=40e-5, # uV (EEG channels)
+                                   eog=250e-5 # uV (EOG channels))
+    tstep : float
+        Length of windows for peak-to-peak detection.
+    decim : int
+        Decimate by choosing only a subsample of data points. Highly
+        recommended for data recorded at high sampling frequencies, as
+        otherwise huge intermediate matrices have to be created and inverted.
+    picks : None | list
+        List of indices of channels to be included. If None, defaults to all
+        MEG and EEG channels.
+    solver : str | function
+        Either a function which takes as its inputs the predictor matrix X
+        and the observation matrix Y, and returns the coefficient matrix b;
+        or a string (for now, only 'pinv'), in which case the solver used
+        is dot(scipy.linalg.pinv(dot(X.T, X)), dot(X.T, Y.T)).T.
+
+    Returns
+    -------
+    ev_dict : dictionary
+        A dictionary where the keys correspond to conditions and the values are
+        Evoked objects with the rE[R/F]Ps. These can be used exactly like any
+        other Evoked object, including e.g. plotting or statistics.
+    """
+
+    if isinstance(solver, string_types):
+        if solver == 'pinv':
+            fast_dot = _get_fast_dot()
+
+            # inv is slightly (~10%) faster, but pinv seemingly more stable
+            def solver(X, Y):
+                return fast_dot(linalg.pinv(fast_dot(X.T, X)),
+                                fast_dot(X.T, Y.T)).T
+        else:
+            raise ValueError("No such solver: {0}".format(solver))
+
+    # prepare raw and events
+    if picks is None:
+        picks = pick_types(raw.info, meg=True, eeg=True, ref_meg=True)
+    info = pick_info(raw.info, picks, copy=True)
+    info["sfreq"] /= decim
+    data, times = raw[picks, ::decim]
+    events = events.copy()
+    events[:, 0] -= raw.first_samp
+    events[:, 0] /= decim
+
+    conds = list(event_id)
+    if covariates is not None:
+        conds += list(covariates)
+
+    # time windows (per event type)
+    if isinstance(tmin, (float, int)):
+        tmin = dict((cond, int(tmin * info["sfreq"])) for cond in conds)
+    else:
+        tmin = dict((cond, int(tmin.get(cond, -.1) * info["sfreq"]))
+                    for cond in conds)
+    if isinstance(tmax, (float, int)):
+        tmax = dict((cond, int((tmax * info["sfreq"]) + 1.)) for cond in conds)
+    else:
+        tmax = dict((cond, int((tmax.get(cond, 1.) * info["sfreq"]) + 1))
+                    for cond in conds)
+
+    # Construct predictor matrix
+    # We do this by creating one array per event type, shape lags * samples
```

Comment:
you mean shape (lags, samples) ? Otherwise, I'll think it's a 1D array you're talking about. Btw, thanks again for extensive documentation.